api/src/controllers/publications.js

```diff
@@ -21,19 +23,23 @@ const { puppeteerConfig, categoryType } = require('../config/puppeteer');
  * @returns 404: publication not found
  * @returns 400: error deleting publication
  */
-async function deletePublication(req, res) {
+async function deletePublication(req, res, next) {
   const { id: _id } = req.params;
-  if (!mongoose.Types.ObjectId.isValid(_id))
-    return res.status(404).send('Error: No publication with that id.');
-
-  try {
-    await Publication.findByIdAndRemove(_id);
-    return res
-      .status(200)
-      .json({ message: 'Publication deleted successfully.' });
-  } catch (err) {
-    res.status(400).json(`Error: ${err.message}`);
-  }
+  await Publication.findByIdAndRemove(_id)
```

Comment:
If using promises instead, I don't think we will need to await a promise and the function no longer need to be async. Same for the other functions as well.

Reply:
I tested it and looks like you're right! I'll change them accordingly then.